backend/src/yeomanui.ts

```diff
@@ -440,35 +454,69 @@ export class YeomanUI {
 
 		const questions: any[] = [];
 
+		let selectedWorkspaceConfig = this.uiOptions.messages.create_and_close;
+		const vscodeInstance = this.getVscode();
+		if (vscodeInstance) {
+			let newSelectedWorkspaceConfig =  await vscodeInstance.workspace.getConfiguration("ApplicationWizard");
+			if (newSelectedWorkspaceConfig){
+				const currentPath = _.get(vscodeInstance, "workspace.workspaceFolders[0].uri.fsPath");
+				if (!currentPath || YeomanUI.PROJECTS.toLowerCase() === currentPath.toLowerCase()){
+					vscodeInstance.workspace.getConfiguration("ApplicationWizard").update("Workspace", this.uiOptions.messages.open_in_a_new_workspace, vscodeInstance.ConfigurationTarget.Global);
+					this.forceNewWorkspace = true;
+					selectedWorkspaceConfig = this.uiOptions.messages.open_in_a_new_workspace;
+				}
+				else if (YeomanUI.PROJECTS.toLowerCase() != currentPath.toLowerCase()) {
+					vscodeInstance.workspace.getConfiguration("ApplicationWizard").update("Workspace", this.uiOptions.messages.add_to_workspace, vscodeInstance.ConfigurationTarget.Global);
+					this.forceNewWorkspace = false;
+					selectedWorkspaceConfig = this.uiOptions.messages.add_to_workspace;
+				}
+			}
+		}
+
 		if (_.includes(genFilter.types, GeneratorType.project)) {
 			const defaultPath = this.getCwd();
+			if (vscodeInstance.workspace.getConfiguration("ApplicationWizard")){
+				vscodeInstance.workspace.getConfiguration("ApplicationWizard").update("TargetFolder", defaultPath, vscodeInstance.ConfigurationTarget.Global);
+			}
```

Comment:
Very strange that preferences will be overwrite on every load of yeoman-ui.
+ @idoprz